components/director/internal/tenantmapping/mapper_for_user.go

```diff
@@ -6,55 +6,110 @@ import (
+		} else {
+			// group name is
+			for _, scope := range group.Scopes {
+
+				if !stringInSlice(scope, filteredScopesArray) {
+					filteredScopesArray = append(filteredScopesArray, scope)
+				}
+			}
+		}
+	}
+
+	return strings.Join(filteredScopesArray, " "), tenantID
 }
 
 func (m *mapperForUser) GetObjectContext(ctx context.Context, reqData ReqData, username string) (ObjectContext, error) {
 	var externalTenantID, scopes string
+	var staticUser StaticUser
+	var staticGroups []StaticGroup
 
-	staticUser, err := m.staticUserRepo.Get(username)
-	if err != nil {
-		return ObjectContext{}, errors.Wrap(err, fmt.Sprintf("while searching for a static user with username %s", username))
-	}
+	userGroups, err := reqData.GetGroups()
+	log.Infof("GetGroups returned %s\n", userGroups)
 
-	scopes, err = reqData.GetScopes()
 	if err != nil {
-		if !apperrors.IsKeyDoesNotExist(err) {
-			return ObjectContext{}, errors.Wrap(err, "while fetching scopes")
-		}
+		return ObjectContext{}, errors.Wrap(err, fmt.Sprintf("while getting groups for a static user with username %s", username))
+	}
 
-		scopes = strings.Join(staticUser.Scopes, " ")
+	if len(userGroups) > 0 {
+		staticGroups = m.staticGroupRepo.Get(userGroups)
 	}
 
-	externalTenantID, err = reqData.GetExternalTenantID()
-	if err != nil {
-		if !apperrors.IsKeyDoesNotExist(err) {
-			return ObjectContext{}, errors.Wrap(err, "while fetching external tenant")
+	if len(staticGroups) > 0 {
+		// proceed with group scopes flow
+		scopes, externalTenantID = GetGroupScopesAndTenant(staticGroups)
+		log.Infof("Decided to use groups copes %s\n", scopes)
+	} else {
+		// proceed with staticUser (and his scopes) flow
+		staticUser, err = m.staticUserRepo.Get(username)
+		if err != nil {
+			return ObjectContext{}, errors.Wrap(err, fmt.Sprintf("while searching for a static user with username %s", username))
 		}
 
-		return NewObjectContext(TenantContext{}, scopes, staticUser.Username, consumer.User), nil
+		scopes, err = reqData.GetScopes()
+		if err != nil {
+			if !apperrors.IsKeyDoesNotExist(err) {
+				return ObjectContext{}, errors.Wrap(err, "while fetching scopes")
+			}
+
+			scopes = strings.Join(staticUser.Scopes, " ")
+			log.Infof("Decided to use staticUser scopes %s\n", scopes)
+		}
+		externalTenantID, err = reqData.GetExternalTenantID()
+		if err != nil {
+			if !apperrors.IsKeyDoesNotExist(err) {
+				return ObjectContext{}, errors.Wrap(err, "while fetching external tenant")
+			}
+			return NewObjectContext(TenantContext{}, scopes, staticUser.Username, consumer.User), nil
+		}
 	}
 
 	tenantMapping, err := m.tenantRepo.GetByExternalTenant(ctx, externalTenantID)
 	if err != nil {
 		return ObjectContext{}, errors.Wrapf(err, "while getting external tenant mapping [ExternalTenantId=%s]", externalTenantID)
 	}
 
-	if !hasValidTenant(staticUser.Tenants, tenantMapping.ExternalTenant) {
+	if len(userGroups) <= 0 && !hasValidTenant(staticUser.Tenants, tenantMapping.ExternalTenant) {
```

Comment:
Why you check here the user groups length? And the length of the slice should never be less than 0.